linux_os/guide/system/network/network-nftables/set_nftables_base_chain/ansible/shared.yml

```diff
@@ -0,0 +1,71 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+#Name of the table
+{{{ ansible_instantiate_variables("var_nftables_table") }}}
+#Familiy of the table 
+{{{ ansible_instantiate_variables("var_nftables_family") }}}
+#Number of base chains
+{{{ ansible_instantiate_variables("var_nftables_base_chain_number") }}}
+#Name(s) of base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_names") }}}
+#Type(s) of base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_types") }}}
+# Hooks for base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_hooks") }}}
+#Priority
+{{{ ansible_instantiate_variables("var_nftables_base_chain_priorities") }}}
+#Policy 
+{{{ ansible_instantiate_variables("var_nftables_base_chain_policies") }}}
+
+- name: Ensure Base Chains Exist for Nftables - Remove file for remediation
```

Comment:
@rumch-se , it should be used `{{{ rule_title }}}` instead of a static title in each task. Please, take a look in the example from our style guide:
https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#ansible

Reply:
Hello @marcusburghardt,
1.There is another CIS rule which gets and after implements nftable configurations from a configuration files.
2.In this case the whole "magic" is in memory. In this version I do a check if base chains exist and I do nothing in this case. I am creating them also if we don't have such chains.
3.According to the file - I was thinking about its location and according to my logic - this file is a temp file and now I delete it (if the file exist), after I create a new one and execute it. But I can delete it after. In this case nothing will remain on the OS.  
4. The main reason to create a file - is because I have defined many of the variables as arrays. The end user can extends these arrays and can create more chains if he wants to. But bash arrays are ("item1" "item2") while ansible arrays are ["item1","item2"] - i.e. to implement the same logic in ansible I have to use varaibles x 2 - one set for bash and second for ansible.
Have a nice day
Rumen